src/App.js

```diff
@@ -7,55 +7,196 @@ import AddTodoForm from './AddTodoForm';
 // View part of react
 const App = () => {
 
+
   const [todoList, setTodoList] = React.useState(
     []
   );
 
+  const [postTodoList, setPostTodoList] = React.useState(
+    []
+  );
+
   const [isLoading, setIsLoading] = React.useState(true);
 
-  React.useEffect(() => {
-    const fetchData = new Promise((resolve, reject) => {
-      setTimeout(() =>{
-        resolve({data: {todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [] }});
-      }, 2000);
-    });
-
-    fetchData
-      .then((result) => {
-        setTodoList(result.data.todoList)
+  const [isError, setIsError] = React.useState(false);
+
+  // GET 
+  const fetchData = async () => {
+
+    setIsLoading(true);
+    
+    const options = {
+      method: "GET",
+      headers: {
+        'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`
+      }
+    }
+    
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/`;
+
+    try {
+      const response = await fetch(url, options);
+      
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const data = await response.json();
+
+      const todos = data.records.map((todo) => {
+        const newTodos = {
+          id: todo.id,
+          title: todo.fields.title,
+        }
+
+        return newTodos
+
       });
 
-    setIsLoading(false);
+      setTodoList(todos)
+
+    } catch (error) {
+      setIsError(error);
+      console.log(error.message)
+    } finally {
+      setIsLoading(false);
+    }
+    
+  };
+
+
+  React.useEffect(() => {
 
+    fetchData();
   }, [])
 
+  const addTodo = (records) => {
+
+    // set the todoList for Get
+    setTodoList([...todoList, records[0].fields]);
+    // set the postTodoList for Post 
+    setPostTodoList({...postTodoList, records});
```

Comment:
@raul-2001 there is no need to maintain 2 to do lists 

When you add a record, send the post to airtable to add the record, then get your items from airtable again to refresh your list using your fetchData function